src/lib/filter.js

```diff
@@ -148,7 +148,7 @@ function formatTerm(uri) {
  * @param {Map<String,Array<String>>} inferredTermsByCategory
  * @returns {String}
  */
-export function generateFilterDescription(searchSettings, inferredTermsByCategory) {
+export function generateFilterDescription(searchSettings,inferredTermsLabelsByCategory,missingLabelsByCategory) {
```

Comment:
The logic here would be much simpler if instead we provided a list of URI and label objects. Unknown labels could default to a formatted URI.

In the future, the endpoints that do ontology inference will include a breakdown of inferred terms. That would avoid querying annotations altogether.

Reply:
Here's the relevant issue https://github.com/PavlidisLab/Gemma/issues/1064